src/cursor.rs

```diff
@@ -473,19 +459,8 @@ impl Cursor {
     ///
     /// Returns a vector containing the BSON documents that were read.
     pub fn next_n(&mut self, n: i32) -> Result<Vec<bson::Document>> {
-        let mut vec = vec![];
-
-        for _ in 0..n {
-            let bson_option = self.next();
-
-            match bson_option {
-                Some(Ok(bson)) => vec.push(bson),
-                Some(Err(err)) => return Err(err),
-                None => break,
-            };
-        }
-
-        Ok(vec)
+        let n = ::std::cmp::max(0, n) as usize;
+        self.take(n).collect()
```

Comment:
Sorry, I'm not sure which error?

Reply:
Oh I think you mean the panic when attempting to construct a range with `0..negative` bounds? I think if you don't mind a breaking change to the API, it would actually be better to make the argument `usize` (i.e. unsigned) up-front instead of accepting a signed integer, because negative values make no sense in this context anyway.

Reply:
yeah, I think taking a usize makes more sense here. cc. @saghm